test/packing/log_auto-test-gen.sh

```diff
@@ -0,0 +1,269 @@
+#!/bin/bash
+
+#
+# This script generates test/packing/log_auto_fmt.h from an older src/log/log_auto.c
+#
+
+set -euo pipefail
+
+GITHOME="$(git rev-parse --show-toplevel)"
+cd "$GITHOME"
+
+export MYNAME=`basename "$0"`
+
+prepare_file() {
+  perl -E '
+    @a=<>;                                   # read all lines
+    for (@a) { /^(?:#define\s+)?(__\w++)(?=\()/ and $fn{$1}=1; }  # collect locally defined functions and fn-like macros
+    $FNRE = join "|", map {"\\b$_\\b"} keys %fn;              # make a regex out of them
+    print << "_END";
+/* DO NOT EDIT: automatically built by $ENV{MYNAME} */
+
+#pragma GCC diagnostic error "-Wno-used-but-marked-unused"
```

Comment:
Because not all functions in the generated code are used. The functions are generated based on the meta-data, it's not easy to select which functions to drop.